docs/adrs/0002-uds-core-functional-layers.md

```diff
@@ -0,0 +1,73 @@
+# ADR: UDS Core Functional Layers
+
+Date: 2024-07-17
+
+## Status
+
+Accepted
+
+## Context
+
+Historically, UDS Core intentionally only published a single official Zarf Package to ensure a standard baseline across all UDS environments. We learned from the complex buffet menu-style of Big Bang that it is better to form opinions and test them than to allow unlimited configuration possibilities. We have also found the monorepo structure of UDS Core to be be much simpler for developing UDS Core. However, through user feedback, we have seen cases where more optionality is warranted, but having to modify UDS CLI to accommodate disabling portions of UDS Core is not ideal. We also have a need for an edge-compatible version of UDS Core that is more lightweight and does not include all the components of the full UDS Core package.
+
+Today, we publish both the official UDS Core package and a UDS Slim Dev package, in addition to dev and demo UDS Bundles. The UDS Slim Dev package is a stripped-down version of UDS Core that includes limited services meant for testing UDS Packages. We have also learned that some teams outside of UDS need similar functionality, but slightly different parts of UDS Core, such as keycloak + authsvc.
+
+## Decision
+
+We will replace the existing 2-package (one dev and one prod) structure with a layered set of packages within the existing monorepo. This will allow users to choose the components they need for their specific use case, while maintaining a single source of truth for all UDS Core components. We will not break these out by applications as Big Bang did, but instead by layers of functionality for related/integrated components. This allows more flexibility for which functionality is provided by UDS Core and which is provided by other external packages.
+
+The functional layers will be:
+
+1. UDS Core Base (Istio + Pepr w/UDS Operator & UDS Policies)
+2. UDS Core Identity and Authorization (Keycloak + AuthService)
+3. UDS Core Metrics (Metrics Server)
+4. UDS Core Monitoring (Prometheus + Grafana)
+5. UDS Core Logging (Promtail + Loki)
+6. UDS Core Runtime Security (NeuVector)
+7. UDS Core Backup and Restore (Velero)
+8. UDS Core (combination of all layers)
```

Comment:
I think we need to be careful about the terminology we use here.  I might consider saving the 'Core' moniker to represent when users have all functional layers.

Reply:
This is intentional because the packages will be `uds-core-*` and `uds-core`, the dependency tracking discussed above will prevent bad behavior

Reply:
Resolving - @rjferguson21 let me know if there are any other concerns here. I think it's good to attribute these layers as part of core rather than generically calling them "UDS Logging". If that does cause confusion in the future I think we should always feel comfortable revisiting decisions and assumptions.